### PR TITLE
Fix place ships bug

### PR DESCRIPTION
**Pull request by lrharris215**
Fixed the original placeShips bug, which did not remove ships from the Player's shipList correctly. Also fixed a validation bug where the Game constructor was not assigning the validators correctly, allowing invalid ship placements to slip through and cause errors.

### CODE REVIEW

File: src/test/java/com/battleship/backend/TestClasses.java [hunk 1]
```diff
@@ -80,6 +80,19 @@ public void addSection(Sectionable section, int row, int col) {
...
+            for(Ship boardShip : shipList){
+                for(Ship otherShip : otherShipList){
+                    if(boardShip.equals(otherShip)){
+                        count += 1;
+                    }
+
+                }
+            }
+            return count == shipList.size();
+        }
```
Comment (kudukacar) [inline src/test/java/com/battleship/backend/TestClasses.java:94]: This method looks very much like the actual method.  Can we simplify for the test?  If the methods are the same why have a test class?

File: src/main/java/com/battleship/backend/models/Board.java [hunk 2]
```diff
@@ -17,6 +17,9 @@ public Board(String name){
     }
 
     public void addShip(Ship ship, int row, int col){
+        System.out.println("ship:" + ship.getName());
+        System.out.println("row: " + row);
+        System.out.println("col: " + col);
```
Comment (Middlebrooks314) [inline src/main/java/com/battleship/backend/models/Board.java]: Was printing to the console part of your requirements? If it was for demo purposes from your mentors, carry on. If not, I'm not so sure this is necessary for the API.

File: src/main/java/com/battleship/backend/models/Board.java [hunk 3]
```diff
@@ -66,6 +69,19 @@ public boolean hasShip(Ship maybeShip){
...
     }
 
+    public boolean shipListEquals(ArrayList<Ship> otherShipList){
+        int count = 0;
+        for(Ship boardShip : shipList){
+            for(Ship otherShip : otherShipList){
+                if(boardShip.equals(otherShip)){
+                    count += 1;
+                }
+
```
Comment (Middlebrooks314) [inline src/main/java/com/battleship/backend/models/Board.java:76]: extra space

File: src/test/java/com/battleship/backend/models/BoardTest.java [hunk 4]
```diff
@@ -78,6 +80,30 @@ void testHasShip(){
...
+
+    @Test
+    void testShipListEqualsReturnsFalseIfNamesDoNotMatch(){
+        ArrayList<Ship> testList = new ArrayList<Ship>();
+        testList.add(new Ship("test1", 2));
+        testList.add(new Ship("test2", 2));
+
+        testBoard.addShip(new Ship("test1", 2), 0 ,0);
+        testBoard.addShip(new Ship("test3", 2), 1 ,0);
+
```
Comment (Middlebrooks314) [inline src/test/java/com/battleship/backend/models/BoardTest.java:103]: Nice 🎉

**Review by Middlebrooks314 — CHANGES_REQUESTED** [review]

File: src/main/java/com/battleship/backend/models/Board.java (hunk 3 above)
Comment (lrharris215) [reply src/main/java/com/battleship/backend/models/Board.java:76]: Removed.

File: src/test/java/com/battleship/backend/models/BoardTest.java (hunk 4 above)
Comment (lrharris215) [reply src/test/java/com/battleship/backend/models/BoardTest.java:103]: Thank you!

File: src/main/java/com/battleship/backend/models/Board.java (hunk 2 above)
Comment (lrharris215) [reply src/main/java/com/battleship/backend/models/Board.java]: Accidentally committed before I removed those. Removed now.

**Review by Middlebrooks314 — APPROVED** [review]

File: src/test/java/com/battleship/backend/validators/PlaceShipsValidatorTest.java [hunk 5]
```diff
@@ -1,9 +1,11 @@
 package com.battleship.backend.validators;
 
 import com.battleship.backend.TestClasses;
+import com.battleship.backend.models.Board;
 import com.battleship.backend.models.Boardable;
 import com.battleship.backend.models.Request;
 import com.battleship.backend.models.Ship;
+import org.apache.logging.log4j.message.ReusableMessage;
```
Comment (seoulection) [inline src/test/java/com/battleship/backend/validators/PlaceShipsValidatorTest.java:8]: 👀  https://www.cnn.com/2021/12/15/tech/log4j-vulnerability/index.html

I don't think using log4j in your project has any repercussions though.

File: src/test/java/com/battleship/backend/models/BoardTest.java [hunk 6]
```diff
@@ -78,6 +80,30 @@ void testHasShip(){
...
+        testList.add(new Ship("test2", 2));
+
+        testBoard.addShip(new Ship("test1", 2), 0 ,0);
+        testBoard.addShip(new Ship("test2", 2), 1 ,0);
+
+        assertTrue(testBoard.shipListEquals(testList));
+    }
+
+    @Test
+    void testShipListEqualsReturnsFalseIfNamesDoNotMatch(){
```
Comment (pumpkincouture) [inline src/test/java/com/battleship/backend/models/BoardTest.java:96]: Nitpick: can we update these two tests to either be:
`testShipListEqualsFalse`
or
`testShipListReturnsFalse`
vs
`testShipListEqualsReturns`

File: src/main/java/com/battleship/backend/models/Board.java [hunk 7]
```diff
@@ -66,6 +66,18 @@ public boolean hasShip(Ship maybeShip){
         return false;
     }
 
+    public boolean shipListEquals(ArrayList<Ship> otherShipList){
```
Comment (seoulection) [inline src/main/java/com/battleship/backend/models/Board.java:69]: Nit: could you please make this adhere to Java's naming convention for booleans?

Comment (lrharris215) [reply src/main/java/com/battleship/backend/models/Board.java:69]: Yes, changed to isShipListEquals.

Comment (lrharris215) [reply src/main/java/com/battleship/backend/models/Board.java:69]: Actually, changed to isPermutationOfShipList

File: src/test/java/com/battleship/backend/models/BoardTest.java (hunk 6 above)
Comment (lrharris215) [reply src/test/java/com/battleship/backend/models/BoardTest.java:96]: The method name was shipListEquals, which led to that unfortunate name. I have changed the name of the method to isPermutationOfShipList, so the updated test names make more sense now.

File: src/test/java/com/battleship/backend/validators/PlaceShipsValidatorTest.java (hunk 5 above)
Comment (lrharris215) [reply src/test/java/com/battleship/backend/validators/PlaceShipsValidatorTest.java:8]: Removed it anyway, wasn't using it. Also removed other unnecessary import statements.

**Pull request merged**